libcontainer/container_linux.go

```diff
@@ -632,7 +632,12 @@ func (c *linuxContainer) newSetnsProcess(p *Process, cmd *exec.Cmd, messageSockP
 			// cgroup v1: using the same path for all controllers.
 			// cgroup v2: the only possible way.
 			for k := range proc.cgroupPaths {
-				proc.cgroupPaths[k] = path.Join(proc.cgroupPaths[k], add)
+				subPath := path.Join(proc.cgroupPaths[k], add)
+				if strings.HasPrefix(subPath, proc.cgroupPaths[k]) {
+					proc.cgroupPaths[k] = subPath
+				} else {
+					return nil, fmt.Errorf("%s is not a sub cgroup path", add)
+				}
```

Comment:
This can be simplified to
```go
if !strings.HasPrefix(subPath, proc.cgroupPaths[k]) {
	return nil, fmt.Errorf("%s is not a sub cgroup path", add)
}
proc.cgroupPaths[k] = subPath
```